nix/build.sh

```diff
@@ -0,0 +1,53 @@
+#!/usr/bin/env bash
+
+set -e
+
+# cleanup for artifacts created during builds
+function cleanup() {
+  # clear trapped signals
+  trap - EXIT ERR INT QUIT
+  # do the actual cleanup
+  ./nix/cleanup.sh ${nixOpts[@]}
+}
+
+trap cleanup EXIT ERR INT QUIT
+
+# build output will end up under /nix, we have to extract it
+function extractResults() {
+  local nixResultPath="$1"
+  echo "Saving build result: ${nixResultPath}"
+  mkdir -p results
+  cp -r "${nixResultPath}/*" results/
```

Comment:
I don't understand how this `results/` folder gets handled later on. Here we seem to be replicating the job of Nix if we were to not pass `--no-out-link`. Who will take care of moving things from `result/` to the expected folder?

Reply:
Well, I haven't implemented that yet because I didn't know what you would think of this approach. We will have to change the CI groovy scripts to use the `results/` folder rather than the current output path for the APK.

And I didn't know that `--no-out-link` was used to block this behavior, that might be cleaner, thanks!